src/main/java/com/google/enterprise/cloudsearch/fs/WindowsFileDelegate.java

```diff
@@ -665,24 +662,13 @@ private void handleChanges(FILE_NOTIFY_INFORMATION info) throws InterruptedExcep
 
       for (ChangeRecord change : changes) {
         log.log(Level.FINER, "Pushing change: " + change);
-        eventPusher.push(change.operation);
+        ListenableFuture<List<GenericJson>> result = eventPusher.push(change.operation);
         try {
-          change.operation.getResult().get();
-        } catch (ExecutionException ex) {
-          if (ex.getCause() instanceof GoogleJsonResponseException) {
-            log.log(Level.FINE, "Caught exception processing change for " + change.path);
-            log.log(Level.FINE,
-                String.valueOf(((GoogleJsonResponseException) ex.getCause()).getDetails()));
-          } else if (ex.getCause() != null
-              && ex.getCause().getCause() instanceof GoogleJsonResponseException) {
-            log.log(Level.FINE, "Caught exception processing change for " + change.path);
-            log.log(Level.FINE,
-                String.valueOf(
-                    ((GoogleJsonResponseException) ex.getCause().getCause()).getDetails()));
-          } else {
-            log.log(Level.FINE, "Caught exception processing change for " + change.path,
-                ex.getCause());
-          }
+          log.log(Level.FINER, "Result for change {0}: {1}",
+              new Object[] { change, result.get(10, TimeUnit.SECONDS) });
+        } catch (ExecutionException | TimeoutException ex) {
+          log.log(Level.FINE, "Caught exception processing change for " + change.path
```

Comment:
Restore the previous logging. I do think that at least single log statements would be preferred, to avoid hard-to-read threading issues.